2018/002-update-channels.md

```diff
@@ -0,0 +1,130 @@
+ Title
+
+| Field  | Value      |
+|:-------|:-----------|
+| Status | Draft      |
+| Date   | 2018-07-31 |
+
+## Introduction
+
+### Problem description
+
+Updates can be disruptive for a customer's application stack (ex. Kubernetes
+upgrade from 1.10.x to 1.11.x, default pod security policy introduction).
+
+Right now there is no visibility of what is inside an update and furthermore
+there is no possible way around intrusive updates by having the ability to only
+apply critical bug fixes and security updates. An admin needs to apply all updates
+that come through a single channel.
+
+In order to separate intrusive updates in SUSE CaaS Platform, we need different
+update channels. This allows the separation of possibly disruptive updates
+into their own channels, and lets the admin continue applying bugfixes and security
+updates without the need to apply intrusive, new features. The admin can also see
+what is inside an update to be able to decide when to apply it.
+
+### Proposed change
+
+The admin will have the ability of choosing when to switch to the next channel.
+A link to the release notes of the update channel is displayed and can be
+evaluated by the admin beforehand.
+
+The UI of velum needs to be adapted to enable a switch to a new channel.
+
+## Detailed RFC
+
+We can separate the updates into different channels. There are 2 types of channels:
+
+* update channel: normal maintenance updates, which are safe and non-intrusive
+* new version channel: version updates which can contain intrusive updates
+
+This RFC proposes:
+
+* a front-end change in the velum UI to enable the admin to switch to a new version
+channel
+* a back-end change in salt to perform the channel switch on all cluster nodes
+
+### Proposed change (Detailed)
+
+#### Tools/Services
+
+We need a set of tools/services to indicate, whenever a maintenance update or a new version
+update is ready to be installed. They should run on each worker node to be more independent
+and decoupled from velum. If velum would be the initiator of the update check, it might lead
```

Comment:
agreed Velum could be a bottleneck but I am not sure how each worker would handle this responsibility. You need to store a binary to check and a manifest *Somewhere* that tells each node what to install. Sounds like Salt (sorry).

Reply:
From SUSE Manager we know that the salt-master, if it has to trigger everything, will become a real bottleneck and not scale for bigger clusters. The easiest way scaling best is really if the OS does the check itself and only reports if there is something to do.
We have all the code, we only need a wrapper around it.